ui/plugin-system/src/components/PluginKindSelect/PluginKindSelect.tsx

```diff
@@ -35,18 +35,18 @@ export const PluginKindSelect = forwardRef((props: PluginKindSelectProps, ref) =
   const { data, isLoading } = useListPluginMetadata(pluginTypes);
 
   // Pass an empty value while options are still loading so MUI doesn't complain about us using an "out of range" value
-  const value = propValue && isLoading ? '' : JSON.stringify(propValue);
+  const value = !propValue || isLoading ? '' : selectionToOptionValue(propValue);
```

Comment:
Please double-check if this is correct.
I don't fully understand why the condition was `propValue && isLoading` earlier (if it's loading, I assume propValue is falsy, and if propValue is defined, the second expression should be called)?